### ISSUE TITLE
bug(Theming Angular Material): Custom typography not applied to standard typography levels in mat.core-theme() or all-component-themes()

### ISSUE THREAD

**Issue by steffenhees** [angular/components#24370]
### Is this a regression?

- [ ] Yes, this behavior used to work in the previous version

### Description

Tried to change the font-family so that e.g. body-1 everywhere has font family Arial.
But it only changed the font-family in mat components, not outside in normal divs.

```
$my-typography: mat.define-typography-config(
  $font-family: 'Arial',
);
$my-theme: mat.define-light-theme((
  typography: $my-typography,
  color: (
    primary: $dicey-primary,
    accent: $dicey-accent,
    warn: $dicey-warn,
  ))
);
@include mat.all-component-themes($my-theme);
```

Similar problem was reported here, but not solved: https://github.com/angular/components/issues/23683#issue-1015421891

-----------------------

It works when I do it with all-component-typographies:

```
$my-typography: mat.define-typography-config(
  $font-family: 'Arial',
);
@include mat.all-component-typographies($my-typography);
```

### Reproduction

Steps to reproduce:
1.  Add custom theme with typography
2.  See if body tag font-family changes

### Expected Behavior

Bodys font family changes to custom font family

### Actual Behavior

The body font family is standard Roboto

### Environment

- Angular: 13.1.2
- CDK/Material: 13.1.1
- Browser(s): Firefox
- Operating System (e.g. Windows, macOS, Ubuntu): Windows 11

**Comment by steffenhees** [comment]
> add the `mat-typography` class to the `body` or some other parent element.
Yes, this is what I have.

Example:
`
<body class="mat-typography">
     <mat-toolbar color="primary"><span>Headline</span></mat-toolbar>
     <div>Some text</div>
</body>
`

The headline had the changed font-family. The "Some text" did not have the changed font family, it had still the default Robot font-family.

**Comment by steffenhees** [comment]
Yes, that are all nice workarounds. I already mentioned that it works with all-component-typographies.
But the bug is, that it is not working with all-component-themes. And this has to be fixed. Please.

I just tried out the kids-theme example on the [website](https://material.angular.io/guide/typography#typography-configs-and-theming) , and this is not working either.

**Comment by crisbeto** [comment]
Technically this works as expected. The `all-component-themes` includes the theme on for Angular Material components, the `mat-typography` classes are considered extra add-ons.

**Comment by umutesen** [comment]
> So I think the problem is that you need to pass in your custom typography to the `mat.core` mixin. Here's a config that worked for me both for the headline and the "Some text" under it.
> 
> ```
> $candy-app-primary: mat.define-palette(mat.$indigo-palette);
> $candy-app-accent: mat.define-palette(mat.$pink-palette, A200, A100, A400);
> $candy-app-theme: mat.define-light-theme((
>   color: (
>     primary: $candy-app-primary,
>     accent: $candy-app-accent
>   ),
>   typography: mat.define-typography-config(
>     $font-family: 'Arial',
>   ),
> ));
> 
> @include mat.core($candy-app-theme);
> @include mat.all-component-themes($candy-app-theme);
> ```
> 
> Alternatively you can include the `all-component-typographies` like you mentioned above since that's what `mat.core` does internally anyway.


`mat-core` does not appear to allow any parameters on v15: 

Module build failed (from ./node_modules/sass-loader/dist/cjs.js):
SassError: Only 0 arguments allowed, but 1 was passed.
  ┌──> src/styles.scss
33│ @include mat.core($my-theme);

**Comment by Judp0m** [comment]
Seems related : https://stackoverflow.com/questions/74595732/angular-material-v15-theme-typography-default-text-not-styled

**Comment by travissimon** [comment]
I've been spending way too much time on this, and have decided that I hope it is a side-effect of the error in compiling the sass files (https://github.com/angular/components/issues/26037).

True, this does seem like a bit of a long shot, but my theory is that the Sass compiler is bailing out when it hits the error, which is why some components receive custom typography styles while others receive the default typography.

If this theory holds true, I saw in another comment that there is going to be a new release of Angular Material in the next 24 hours or so that will hopefully resolve this issue.

**Comment by maxime1992** [comment]
`@include mat.core($typography);` how about the `mat.legacy-core` ? It doesn't seem to take any argument and I've got parts of the app where the font isn't properly taken into account anymore

**Comment by Kiriakos41** [comment]
any fix?

**Comment by mmalerba** [comment]
I think maybe there are multiple things being asked for in this issue:
1. Setting the core angular material typography styles (for things like mat-option) should be down with `mat.core-theme($theme)` or `mat.core-typography($typography-config)`
2. Setting the typography for things like h1, h2, etc. should be down with `mat.typography-hierarchy($theme)`

I'm going to mark this as a docs issue, because I think the section on typography is out of date and needs an update to explain these different mixins and what they're responsible for.

**Comment by MahmoudTarek44** [comment]
Is it possible that we can change custom font family on [dir] change ??
for example:
ltr > Poppins
rtl > Tajawal

**Comment by temperature301** [comment]
> > add the `mat-typography` class to the `body` or some other parent element.
> > Yes, this is what I have.
> 
> Example: `<body class="mat-typography">` `<mat-toolbar color="primary"><span>Headline></span></mat-toolbar>` `<div>Some text </div>` `</body>`
> 
> The headline had the changed font-family. The "Some text" did not have the changed font family, it had still the default Robot font-family.





https://angulars.space/questions/the-themes-from-infragistics-are-not-functioning-as-anticipated-the-default-theme-is-not-being-applied-properly
